tests/SideBySide.New/StoredProcedureTests.cs

```diff
@@ -0,0 +1,149 @@
+CREATE PROCEDURE simple()
+BEGIN
+	SELECT 'simple';
+END";
+				await cmd.ExecuteNonQueryAsync();
+			}
+
+			using (var cmd = m_database.Connection.CreateCommand())
+			{
+				cmd.CommandText = "simple";
+				cmd.CommandType = CommandType.StoredProcedure;
+				cmd.Parameters.Add(new MySqlParameter
+				{
+					ParameterName = "@result",
+					DbType = DbType.String,
+					Direction = ParameterDirection.ReturnValue,
+				});
+				await cmd.ExecuteNonQueryAsync();
+				Assert.Equal("simple", cmd.Parameters["@result"].Value);
+			}
+		}
+
+		[Fact]
+		public async Task StoredProcedureComplex()
+		{
+			using (var cmd = m_database.Connection.CreateCommand())
+			{
+				cmd.CommandText = @"DROP PROCEDURE IF EXISTS complex;
+CREATE PROCEDURE complex(
+	IN radius DOUBLE,
+	IN height DOUBLE,
+	IN name VARCHAR(63),
+	OUT diameter DOUBLE,
+	OUT circumference DOUBLE,
+	OUT area DOUBLE,
+	OUT volume DOUBLE,
+	OUT shape VARCHAR(63)
+)
+BEGIN
+	SELECT radius * 2 INTO diameter;
+	SELECT diameter * PI() INTO circumference;
+	SELECT PI() * POW(radius, 2) INTO area;
+	SELECT area * height INTO volume;
+	SELECT 'circle' INTO shape;
+	SELECT CONCAT(name, shape);
+END";
+				await cmd.ExecuteNonQueryAsync();
+			}
+
+			using (var cmd = m_database.Connection.CreateCommand())
+			{
+				cmd.CommandText = "complex";
+				cmd.CommandType = CommandType.StoredProcedure;
+				cmd.Parameters.Add(new MySqlParameter
+				{
+					ParameterName = "@radius",
+					DbType = DbType.Double,
+					Direction = ParameterDirection.Input,
+					Value = (double)1,
+				});
+				cmd.Parameters.Add(new MySqlParameter
+				{
+					ParameterName = "@height",
+					DbType = DbType.Double,
+					Direction = ParameterDirection.Input,
+					Value = (double)2,
+				});
+				cmd.Parameters.Add(new MySqlParameter
+				{
+					ParameterName = "@name",
+					DbType = DbType.String,
+					Direction = ParameterDirection.Input,
+					Value = "awesome",
+				});
+				cmd.Parameters.Add(new MySqlParameter
+				{
+					ParameterName = "@diameter",
+					DbType = DbType.Double,
+					Direction = ParameterDirection.Output,
+				});
+				cmd.Parameters.Add(new MySqlParameter
+				{
+					ParameterName = "@circumference",
+					DbType = DbType.Double,
+					Direction = ParameterDirection.Output,
+				});
+				cmd.Parameters.Add(new MySqlParameter
+				{
+					ParameterName = "@area",
+					DbType = DbType.Double,
+					Direction = ParameterDirection.Output,
+				});
+				cmd.Parameters.Add(new MySqlParameter
+				{
+					ParameterName = "@volume",
+					DbType = DbType.Double,
+					Direction = ParameterDirection.Output,
+				});
+				cmd.Parameters.Add(new MySqlParameter
+				{
+					ParameterName = "@shape",
+					DbType = DbType.String,
+					Direction = ParameterDirection.Output,
+				});
+				cmd.Parameters.Add(new MySqlParameter
+				{
+					ParameterName = "@result",
+					DbType = DbType.String,
+					Direction = ParameterDirection.ReturnValue,
+				});
+				await cmd.ExecuteNonQueryAsync();
```

Comment:
Again, I think you need to `ExecuteReaderAsync` here and read the (single) row that is returned.